src/nni_manager/training_service/kubeflow/kubeflowTrainingService.ts

```diff
@@ -139,22 +138,26 @@ class KubeflowTrainingService implements TrainingService {
 
             await fs.promises.writeFile(path.join(trialLocalTempFolder, 'run_worker.sh'), workerRunScriptContent, { encoding: 'utf8' });
         }
-
         // Write parameter server file content run_ps.sh to local tmp folders
-        if(this.kubeflowTrialConfig.ps) {
+        if(this.kubeflowTrialConfig instanceof KubeflowTrialConfigTensorflow && this.kubeflowTrialConfig.ps) {
```

Comment:
It's good to use this.kubeflowClusterConfig.operator to explicitly convet this.kubeflowTrialConfig to specific sub-class's instance